code/DDSCodeTester.cpp

```diff
@@ -2406,10 +2422,20 @@ class CustomDataWriterListener : public DataWriterListener
             DataWriter* writer,
             const LivelinessLostStatus& status)
     {
-        (void)writer, (void)status;
+        static_cast<void>(writer);
+        static_cast<void>(status);
         std::cout << "Liveliness lost. Matched Subscribers will consider us offline" << std::endl;
     }
 
+    virtual void on_unacknowledged_sample_removed(
+            DataWriter* writer,
+            const InstanceHandle_t& instance)
```

Comment:
I would like the CI to fail if we break the API in the future...

```suggestion
    void on_unacknowledged_sample_removed(
            DataWriter* writer,
            const InstanceHandle_t& instance) override
```